### PR TITLE
Upgraded tools; Split of CI jobs into pieces; Moved some to GH actions; different way of managing tools.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

Fixes: https://github.com/thanos-io/thanos/issues/2663

### CODE REVIEW

**Comment by bwplotka** [comment]
This should be ready for review :hugs: 

Finally after a day of fighting with this. Now most of the things are split, plus golint no longer OOMs (upgrade + move to GH actions)

I also used my https://github.com/bwplotka/bingo project for binary management. This however is not needed for *USAGE* of those binaries, just for managing them. I will write small blog post about it.

**Comment by bwplotka** [comment]
OK, Moving doc checks back to .CircleCI  GH action constantly fails egress to outside website for LICHE. Let's figure out that in another PR.

**Comment by bwplotka** [comment]
Green!

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: go
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+  pull_request:
+
+# TODO(bwplotka): Add tests here.
```
Comment (pracucci) [inline .github/workflows/go.yaml:10]: e2e tests or unit tests? The first one has been done.

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      - uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Format.
+        run: make format
+
+      - name: Linting & vetting.
+        run: make go-lint
```
Comment (pracucci) [inline .github/workflows/go.yaml:56]: Note to myself: before we were calling `make lint` which calls `go-lint` and `react-app-lint` targets. Now, calling only `go-lint`, we've lost `react-app-lint`, but `react-app-lint` is called by `react-app-test` in the react workflow, so all good.

File: .bingo/Variables.mk [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+#
+# In your main Makefile (for non array binaries):
+#
+#include .bingo/Variables.mk # (If not generated automatically by bingo).
+#
+#command: $(ALERTMANAGER)
+#	@echo "Running alertmanager"
+#	@$(ALERTMANAGER) <flags/args..>
+#
+ALERTMANAGER ?= $(GOBIN)/alertmanager-v0.20.0
```
[the remark this answers is not in the document]
Comment (brancz) [reply .bingo/Variables.mk:18]: Since this is a generated file, I don't think it's too bad, but agreed.

**Comment by brancz** [comment]
Gonna go ahead and merge this, to make CI work on master again.

File: .bingo/Variables.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (bwplotka) [reply .bingo/Variables.mk:18]: No need for this, https://github.com/bwplotka/bingo recreates this file anyway (:

File: .github/workflows/go.yaml (hunk 1 above)
Comment (bwplotka) [reply .github/workflows/go.yaml:10]: Unit tests (: 

They are still in the Circle CI for now.

File: .github/workflows/go.yaml (hunk 2 above)
Comment (bwplotka) [reply .github/workflows/go.yaml:56]: yes (:

**Pull request merged**